vispy/visuals/mesh.py

```diff
@@ -357,16 +424,77 @@ def _update_data(self):
                 normals = (1., 0., 0.)
 
             self.shared_program.vert['normal'] = normals
-            self.shared_program.vert['base_color'] = colors
+            self.shared_program.vert['color_transform'] = \
+                _build_color_transform(colors, self._cmap, self._clim_values)
+            if colors.ndim == 1:
+                self.shared_program.vert['base_color'] = colors
+            else:
+                self.shared_program.vert['base_color'] = VertexBuffer(colors)
```

Comment:
and a conditional based on whether we're doing the value->color conversion with colormap, or the old/standard way (per-vertex colors actually passed in)